File: kotlin/yass/test/ch/softappeal/yass/javadir/serialize/contract/FieldModifiers.java
package ch.softappeal.yass.javadir.serialize.contract;

public class FieldModifiers {

    public static boolean CONSTRUCTOR_CALLED;

    public static int STATIC_FIELD;

    public transient int transientField;

    private int privateField;

    private int privateFinalField;

    public int publicField;

    public final int publicFinalField;

    private FieldModifiers() {
        CONSTRUCTOR_CALLED = true;
        privateFinalField = 100;
        publicFinalField = 101;
    }

}

File: kotlin/yass/test/ch/softappeal/yass/serialize/FieldModifiers.kt
package ch.softappeal.yass.serialize

var CONSTRUCTOR_CALLED: Boolean = false

class FieldModifiers private constructor() {
    private val privateField: Int = 0
    private val privateFinalField: Int
    var publicField: Int = 0
    val publicFinalField: Int

    init {
        CONSTRUCTOR_CALLED = true
        privateFinalField = 100
        publicFinalField = 101
    }
}
